### DISCUSSION TITLE
Добавить аутентификация логином и паролем

### DISCUSSION THREAD

**Discussion by vikarh** [hufrea/byedpi#186] [Ideas]
Добавить аутентификация пользователя логином и паролем? Для доступа снаружи на андроид при использовании SocksDroid.

**Comment by MeteoraCD2** [comment]
И не только на android. Поддерживаю необходимость авторизации

**Comment by jura1243** [comment]
народ пишет "you can use 3proxy in front of byedpi and setup auth there" . я не проверял. https://github.com/3proxy/3proxy?tab=readme-ov-file

**Reply by jura1243** [reply]
я не проверял но ИИ выдал такой конфиг для /etc/3proxy.cfg:
```
# Запуск в фоне
daemon

# Логирование
log /var/log/3proxy.log D
logformat "- +L%t.%. %N[%P] %e{User-Agent} %U %C"
rotate 30

# Ограничение соединений
maxconn 100

# Работает от имени пользователя (по желанию)
user nobody

# Включить аутентификацию (если нужно на вашем 3proxy)
auth strong
users user1:CL:yourpassword

# Ваш локальный прокси (например, HTTP)
proxy -p3128 -i127.0.0.1

# Upstream (внешний) прокси: все запросы пойдут через него
# Формат: parent protocol port upstream_host upstream_port [auth]
parent 1000 http 8080 proxy.external.com 8080 login:password
```

**Reply by jura1243** [reply]
Объяснение ключевой строки:
`parent 1000 http 8080 proxy.external.com 8080 login:password`

1000 — таймаут в секундах
http — тип протокола (можно socks4, socks5, connect, socks, https)
8080 — порт, который слушает ваш локальный 3proxy (-p8080)
proxy.external.com 8080 — адрес и порт внешнего прокси
login:password — учетные данные для внешнего прокси (если требуется)

**Comment by geeqla** [comment]
Хотелось бы поддержку "аутентификации из коробки", но автор уже где-то сказал, что её не будет (программа предназначена для запуска внутри локальной сети). Если нужен доступ извне, рекомендую https://github.com/ginuerzh/gost.

Как это работает:

1. Gost стартует на порту 1080 и использует файл secrets.txt, в котором хранятся данные пользователей.
2. Gost перенаправляет все запросы на внутренний порт 6666 (или любой другой), где бежит byedpi.
3. Наружу торчит только порт 1080. Подключиться напрямую к byedpi нельзя.

Сборка (требуется Go):
> git clone https://github.com/ginuerzh/gost.git
cd gost/cmd/gost
go build

Старт:
`./gost -L=socks5://0.0.0.0:1080?secrets=/path/to/secrets.txt -F=socks5://0.0.0.0:6666`

Файл secrets.txt:
user1 qwerty111
user2 pass222

Byedpi:
`./ciadpi --ip 0.0.0.0 --port 6666 <параметры>`

**Reply by jura1243** [reply]
с 3proxy у меня не получилось. но я решил ситуацию по другому. подключаюсь к домашней сети по vpn с логином и паролем. потом указываю локальный адрес прокси на подключенной машине.
